src/Razor/test/Microsoft.VisualStudio.LanguageServices.Razor.Test/Cohost/CohostDocumentCompletionEndpointTest.cs

```diff
@@ -484,11 +502,9 @@ private async Task VerifyCompletionListAsync(
         var requestInvoker = new TestLSPRequestInvoker([(Methods.TextDocumentCompletionName, response)]);
 
         var snippetCompletionItemProvider = new SnippetCompletionItemProvider(new SnippetCache());
-        if (snippetLabels is not null)
-        {
-            var snippetInfos = snippetLabels.Select(label => new SnippetInfo(label, label, label, string.Empty, SnippetLanguage.Html)).ToImmutableArray();
-            snippetCompletionItemProvider.SnippetCache.Update(SnippetLanguage.Html, snippetInfos);
-        }
+        snippetCompletionItemProvider.SnippetCache.Update(SnippetLanguage.Html, [
+            new SnippetInfo("snippet1", "snippet1", "snippet1", string.Empty, SnippetLanguage.Html),
+            new SnippetInfo("snippet2", "snippet2", "snippet2", string.Empty, SnippetLanguage.Html)]);
```

Comment:
I really don't like this actually. It's not at all clear not whey the test is specifying that "snippet1" and "snippet2" are not expected. Also we are hardcoding setup in Verify method. 

I wouldn't have a problem with some sort of a common setup instead, but this looks wrong to me.

Reply:
All of our test setup is in this Verify method, and this change is simply making snippets part of the common setup. IMO tests should state the inputs, and expected outputs, and only do set up if there is something special they need. In a real user scenario there are snippets that are installed by default, therefore having snippets present by default in tests makes sense to me. If there was a test that specifically wanted to cover a scenario where there are no snippets, that should be the exception not the rule.

I've updated the tests to use a shared constant for each snippet, so there is no "magic string" effect in the tests, does that help?

Reply:
Thank you, that does make it easier to reason about it IMHO - I may have even made the snippet label collection private static readonly field. 

I guess a logical question would be - should be we do the same thing with HTML labels then? (Not asking you to do either change in this PR)

Reply:
Yeah, probably. If we assume that any time Html is asked for completion items it will return "div" and "a", then a test that doesn't specify `delegatedItemLabels` could be passing by accident because it assumes that the delegated server wasn't called, but in actual fact there just wasn't a test setup for it.

Probably should have all tests explicitly verify whether they expect Html completion, C# completion, tag helper completion and/or snippet completion, and verify the presence/lack of items every time.

Reply:
I ended up reverting this change, as it was actually hiding a bug. I added it initially because I was annoyed at a test failing because the "Update" method on the snippet provider happened to not be called, and I thought it was a test setup issue, but turns out that actually signifies a product bug and should have failed the test, just ideally in a better way. I'm fixing that now by adding dummy delegated and snippet items that should never appear as results, and failing the test if they do.